devito/passes/iet/mpi.py

```diff
@@ -92,13 +93,15 @@ def rule1(dep, candidates):
         for hs in halo_spots:
             hsmapper[hs] = hs.halo_scheme
 
-            for f in hs.fmapper:
+            for f, (loc_indices, _) in hs.fmapper.items():
                 for n, i in enumerate(iters):
                     candidates = set().union(*[i.dim._defines for i in iters[n:]])
+                    loc_dims = frozenset().union([q for d in loc_indices
```

Comment:
This can be moved outside of this inner loop -- and thus be computed once and for all -- as soon as `loc_indices` is available? ie, first statement of the `for f, (loc_indices, ...` loop

Reply:
My bad!!